File: agent/library/src/main/java/org/matilda/logger/CompoundLogger.java
package org.matilda.logger;

import java.util.Arrays;

public class CompoundLogger implements Logger {
    private final Logger[] mLoggers;

    public CompoundLogger(Logger... loggers) {
        mLoggers = loggers;
    }

    @Override
    public void log(String message) {
        Arrays.stream(mLoggers).forEach(logger -> logger.log(message));
    }

    @Override
    public void log(String message, Throwable throwable) {
        Arrays.stream(mLoggers).forEach(logger -> logger.log(message, throwable));
    }
}

File: agent/library/src/commonMain/kotlin/org/matilda/logger/CompoundLogger.kt
package org.matilda.logger

class CompoundLogger(vararg loggers: Logger) : Logger {
    private val mLoggers = loggers
    override fun log(message: String) {
        mLoggers.forEach { it.log(message) }
    }

    override fun log(message: String, throwable: Throwable) {
        mLoggers.forEach { it.log(message, throwable) }
    }
}
